Data/ActivityRepository.cs

```diff
@@ -16,5 +16,18 @@ public static List<ISportActivity> GetAll()
             return _activityList;
         }
 
+        public List<SportActivity> GetActivitiesByDate(DateTime dateOfActivity)
+        {
+            List<SportActivity> activitiesByDate = new List<SportActivity>();
+            foreach (SportActivity activity in _activityList)
+            {
+                if (activity.Date.Date == dateOfActivity.Date)
```

Comment:
You need to make sure that comparing Dates by the equality operator == works here. Sometimes if Date values differ by  time values it can be problematic. Did you test this code?